dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt

```diff
@@ -0,0 +1,131 @@
+ *
+ * ### Usage Example:
+ * ```kotlin
+ * val keyManager = InMemoryKeyManager()
+ * val did = StatefulWebDid("did:web:tbd.website", keyManager)
+ * ```
+ */
+public class StatefulWebDid(
+  uri: String,
+  keyManager: KeyManager
+) : Did(uri, keyManager)
+
+/**
+ * Configuration options for the [DidWebApi].
+ *
+ * - [engine] is used to override the default ktor engine, which is [CIO].
+ */
+public class DidWebApiConfiguration internal constructor(
+  public var engine: HttpClientEngine? = CIO.create { },
+)
+
+/**
+ * Returns a [DidWebApi] instance after applying [blockConfiguration].
+ */
+public fun DidWebApi(blockConfiguration: DidWebApiConfiguration.() -> Unit): DidWebApi {
+  val conf = DidWebApiConfiguration().apply(blockConfiguration)
+  return DidWebApiImpl(conf)
+}
+
+private class DidWebApiImpl(configuration: DidWebApiConfiguration) : DidWebApi(configuration)
+
+private const val wellKnownURLPath = ".well-known/"
+private const val didDocFilename = "did.json"
+
+/**
+ * Implements [resolve] and [create] according to https://w3c-ccg.github.io/did-method-web/
+ */
+public sealed class DidWebApi(
+  configuration: DidWebApiConfiguration
+) : DidMethod<StatefulWebDid, CreateDidOptions> {
+
+  private val mapper = jacksonObjectMapper()
+
+  private val engine: HttpClientEngine = configuration.engine ?: CIO.create {}
+
+  private val client = HttpClient(engine) {
+    install(ContentNegotiation) {
+      jackson { mapper }
+    }
+  }
+
+  override val methodName: String = "web"
+
+  override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
+    val docURL = getDocURL(did)
+
+    val resp = runBlocking {
+      client.get(docURL) {
+        contentType(ContentType.Application.Json)
+      }
+    }
+
+    val body = runBlocking { resp.bodyAsText() }
+
+    if (!resp.status.isSuccess()) {
+      throw InvalidStatusException(resp.status.value, "resolution error response: '$body'")
+    }
+    return DidResolutionResult(
+      didDocument = mapper.readValue(body, DIDDocument::class.java),
+    )
+  }
+
+  private fun getDocURL(didWebStr: String): String {
+    val parsedDid = DID.fromString(didWebStr)
+    require(parsedDid.methodName == methodName) {
+      "$didWebStr is missing prefix \"did:$methodName\""
+    }
+
+    val subStrs = parsedDid.methodSpecificId.split(":")
+
+    val decodedDomain = URLDecoder.decode(subStrs[0], StandardCharsets.UTF_8)
+
+    return if (subStrs.size == 1) {
+      "https://$decodedDomain/$wellKnownURLPath$didDocFilename"
+    } else {
+      val urlBuilder = StringBuilder()
+      urlBuilder.append("https://$decodedDomain/")
+      for (i in 1 until subStrs.size) {
+        val str = URLDecoder.decode(subStrs[i], StandardCharsets.UTF_8)
+        urlBuilder.append("$str/")
+      }
+      urlBuilder.append(didDocFilename)
+      urlBuilder.toString()
+    }
+  }
+
+  public override fun create(keyManager: KeyManager, options: CreateDidOptions?): StatefulWebDid {
```

Comment:
We can create the json files needed for them to host  at their endpoints. (like this - https://identity.foundation/.well-known/did-configuration.json)

We should add this functionality eventually if not in this pr

Reply:
Did you mean https://identity.foundation/.well-known/did.json ? If so, yes, I think that makes sense. That's similar to what we did in ssi-service